include/xtensor/xmath.hpp

```diff
@@ -2307,6 +2307,42 @@ namespace detail {
             }
         };
 
+        struct nan_min
+        {
+            template <class T, class U>
+            constexpr auto operator()(const T lhs, const U rhs) const
+                -> decltype(math::minimum<void>()(lhs, rhs))
```

Comment:
I don't understand why the CI with gcc4 failed here. I guess it is something related to return type deduction but don't know why.